.github/workflows/contracts.yml

```
@@ -34,6 +34,24 @@ jobs:
           FORKING_URL: ${{ secrets.MAINNET_ETH_HOSTNAME }}
         run: yarn test:system
 
+  contracts-deployment-dry-run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+
+      - name: Cache yarn directory
+        uses: c-hive/gha-yarn-cache@v1
```

We could try using this functionality in `setup-node` action for caching:
```suggestion
      - uses: actions/setup-node@v2
        with:
          node-version: "12.x"
          cache: "yarn"
```
(Also applies to other jobs in the workflow)